accumulator/forest.go

```diff
@@ -113,39 +113,40 @@ type Forest struct {
 //               forest types though (meaning there isn't functionality implemented
 //               to convert a CowForest to DiskForest and vise-versa). Pass a filepath
 //               and cowMaxCache(how much MB to use in ram) to create a CowForest.
-func NewForest(forestFile *os.File, cached bool,
-	cowPath string, cowMaxCache int) *Forest {
+
+type ForestName int
+
+const (
+	DiskForest ForestName = iota
+	RamForest
+	CacheForest
+	CowForest
+)
+
+func (fn ForestName) NewForest(forestFile *os.File, cowPath string, cowMaxCache int) *Forest {
```

Comment:
All exported functions in Go should have a comment with the format:

```
// FunctionName is ...
```

This is how go documentation is done. The documentation is automatically parsed and for package `accumulator`, it is at https://pkg.go.dev/github.com/mit-dci/utreexo/accumulator

Reply:
Also, the function `NewForest()` should not be a method on `ForestName`. It should accept a `ForestName` (`ForestName` should be renamed to `ForestType`) value as a function argument.